src/events/member_async_handler.h

```diff
@@ -0,0 +1,27 @@
+//
+// Created by Piotr Moszkowicz on 2019-02-12.
+//
+
+#pragma once
+#include <future>
+
+#include "event.h"
+#include "function_handler.h"
+
+namespace Bald::Events {
+    template<class T, class EventType>
+    class MemberAsyncHandler : public FunctionHandler<T, EventType> {
+    public:
+        typedef void (T::*HandlerFunction)(EventType*);
+
+        constexpr MemberAsyncHandler(T* instance, HandlerFunction handler_function) noexcept : FunctionHandler<T, EventType>(instance, handler_function) {};
+    private:
+        constexpr void call(Event* e) const noexcept override {
+            auto handle = std::async([&](EventType* params) {
```

Comment:
What do you capture here?

Reply:
Do you mean what handle is? It returns future.

Reply:
no no, I mean why is it [&] there

Reply:
I need to have this context inside of lambda, without it, it errors. By & I send this context into the lambda.